### ISSUE TITLE
Internal Server error, no logs

### ISSUE THREAD

**Issue by sglebs** [tsuru/redisapi#28]
Hi,

I am using https://github.com/tsuru/tsuru-bootstrap.git  and decided to try redisapi as well. For that, I installed Redis on the tsuru-bootstrap VM itself:

ssh -l vagrant 192.168.46.226
sudo apt-get install redis-server
ifconfig | grep docker , get IP of this ifc , got 172.17.42.1
sudo cp /etc/redis/redis.conf /etc/redis/redis.conf.original
sudo vi /etc/redis/redis.conf , change bind: 127.0.0.1 172.17.42.1
tsuru env-set -a redisapi REDIS_SERVER_HOST=172.17.42.1
cp service.yaml service.namoroon.yaml
adjust url: redisapi.192.168.46.226.nip.io , also add password line
crane create service.namoroon.yaml
tsuru service-add redisapi namoroonredis small -t namoroon

At this point I get this:

Error: Failed to create the instance namoroonredis: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>

I tried this to no avail: tsuru env-set -a redisapi LOGLEVEL=DEBUG

So, the obvious questions are:

1) how can I enable more logging so I can troubleshoot?
2) Any idea why I can't create the service instance?

   If I can suggest something, I'd say the documentation could be improved <grin>. It is not clear to the reader if the Redis instances are managed by redisapi itself, via docker instances, or if I really need to pre-install a shared Redis server like I am doing. The explanation of REDIS_SERVER_HOST led me to believe that I do need to pre-install one (which in itself raised confusion in my head: how can I have N redis instances like this then???).

Thanks!

**Comment by fsouza** [comment]
The README is _really_ outdated, this API does support creating new instances of Redis on demmand, but it requires at least one Docker node, and some sentinel hosts (you need at least 3), because it creates containers for running Redis and manage multi-master environments with [sentinel](http://redis.io/topics/sentinel). So, you need:

```
% tsuru env-set -a redisapi SENTINEL_HOSTS='["192.168.1.203:26379", "192.168.1.222:26379", "192.168.2.205:26379", "192.168.2.204:26379", "192.168.2.203:26379"]'
% tsuru env-set -a redisapi DOCKER_HOSTS='["http://192.168.1.23:2375", "https://192.168.1.25:2376"]' REDIS_IMAGE=tsuru/redis
```

REDIS_IMAGE is the Docker image that will be used. We provide a public image in the Docker Hub.

redisapi also uses MongoDB for storing metadata information about the instances it creates, so you need to export some environment variables related to MongoDB:

```
% tsuru env-set -a redisapi MONGODB_URI="mongodb://user:password@mongodb:27017/redisapi DATABASE_NAME=redisapi"
```

Last, but not least, you need to specify which plans are going to be available for users of the API. There are two options: basic and plus. The basic plan doesn't offer high availability and creates only one container for the instance, and the plus offers high availability with multiple containers and sentinel:

```
% tsuru env-set -a redisapi REDIS_API_PLANS='["basic",  "plus"]'
```

Please let me know if this works for you!

**Comment by sglebs** [comment]
Do I really need the docker node and sentinels for the basic plan? Remember, row now I want to test tsuru for DEV deploys on my own laptop. Same for the mongo stuff? Otherwise I might just manually create 2 docker instances that I need for the 2 redis servers that I need for the 2 apps I want to test locally.

BTW, how can I simply create a docker image inside tsuru? If I wanted to add 2 docker instances, is that possible? Thanks!

**Comment by fsouza** [comment]
@sglebs unfortunately yes :-( even for the basic plan you need the full "stack".

Regarding the docker image question, I didn't quite understand it. Do you mean an arbitrary Docker image?

**Comment by sglebs** [comment]
Yes, I just want to manually add 2 redis instances as docker images. This redisapi plugin is too much rocket science (and overkill) for what I need (DEV setup to locally deploy and test). I guess I can do that to the enclosing vagrant vm of tsuru-bootstrap but I was wondering if tsuru supports adding/removing plain docker instances too. I seem to have seen it somewhere in the docs :-)

**Comment by fsouza** [comment]
You can run the containers yourself, but tsuru won't see the containers as members of any app. Currently, tsuru doesn't support running Docker images directly, but it's a common feature request, and we're probably going to implement that in the future.
